roles/dg-config/tasks/main.yml

```diff
@@ -0,0 +1,198 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+- name: DG | SQL to add standby logs
+  set_fact:
+    sql_for_dg:  |
+      alter system set dg_broker_start = true;
+      alter system set standby_file_management = auto;
+      begin
+      for c in (
+        select bytes, thr, (max(cnt_log) - nvl(max(cnt_sby), 0)) diff from (
+        select bytes, thread# thr, count(*) cnt_log, null cnt_sby from v\$log where thread# = 1 group by bytes, thread#
+        union all
```

Comment:
Assuming you want to avoid recreating existing standby redo logs, should this be a "minus"?

Reply:
(max(cnt_log) - nvl(max(cnt_sby), 0)) diff column is using to calculate difference between numbers of logs for thread (and actually size of logs) 
when no standby logs loop will create +1 groups of them for each thread
if they already exist - diff will be -1 avoiding the loop to be called

db12> select * from  v$log;

    GROUP#    THREAD#  SEQUENCE#      BYTES  BLOCKSIZE    MEMBERS ARC STATUS           FIRST_CHANGE# FIRST_TIM NEXT_CHANGE# NEXT_TIME     CON_ID
---------- ---------- ---------- ---------- ---------- ---------- --- ---------------- ------------- --------- ------------ --------- ----------
         1          1         13  104857600        512          2 YES ACTIVE                 1616137 10-JUL-20      1638880 10-JUL-20          0
         2          1         14  104857600        512          2 YES ACTIVE                 1638880 10-JUL-20      1659605 10-JUL-20          0
         3          1         15  104857600        512          2 NO  CURRENT                1659605 10-JUL-20   1.8447E+19                    0

Elapsed: 00:00:00.02
db12> select * from v$standby_log;

    GROUP# DBID                                        THREAD#  SEQUENCE#      BYTES  BLOCKSIZE       USED ARC STATUS     FIRST_CHANGE# FIRST_TIM NEXT_CHANGE# NEXT_TIME LAST_CHANGE# LAST_TIME     CON_ID
---------- ---------------------------------------- ---------- ---------- ---------- ---------- ---------- --- ---------- ------------- --------- ------------ --------- ------------ --------- ----------
         4 UNASSIGNED                                        1          0  104857600        512          0 YES UNASSIGNED                                                                                0
         5 UNASSIGNED                                        1          0  104857600        512          0 YES UNASSIGNED                                                                                0
         6 UNASSIGNED                                        1          0  104857600        512          0 YES UNASSIGNED                                                                                0
         7 UNASSIGNED                                        1          0  104857600        512          0 YES UNASSIGNED                                                                                0


db12> select bytes, thr, (max(cnt_log) - nvl(max(cnt_sby), 0)) diff from (
        select bytes, thread# thr, count(*) cnt_log, null cnt_sby from v$log group by bytes, thread#
        union all
        select bytes, thread# thr, null cnt_log, count(*) cnt_sby from v$standby_log group by bytes, thread#
        )
        group by bytes, thr
/
  2    3    4    5    6    7
     BYTES        THR       DIFF
---------- ---------- ----------
 104857600          1         -1


db12> alter database add logfile thread 2 size 100m;

Database altered.

Elapsed: 00:00:00.53
db12> select * from  v$log;

    GROUP#    THREAD#  SEQUENCE#      BYTES  BLOCKSIZE    MEMBERS ARC STATUS           FIRST_CHANGE# FIRST_TIM NEXT_CHANGE# NEXT_TIME     CON_ID
---------- ---------- ---------- ---------- ---------- ---------- --- ---------------- ------------- --------- ------------ --------- ----------
         1          1         13  104857600        512          2 YES INACTIVE               1616137 10-JUL-20      1638880 10-JUL-20          0
         2          1         14  104857600        512          2 YES INACTIVE               1638880 10-JUL-20      1659605 10-JUL-20          0
         3          1         15  104857600        512          2 NO  CURRENT                1659605 10-JUL-20   1.8447E+19                    0
         8          2          0  104857600        512          2 YES UNUSED                       0                      0                    0

Elapsed: 00:00:00.01
db12> select bytes, thr, (max(cnt_log) - nvl(max(cnt_sby), 0)) diff from (
        select bytes, thread# thr, count(*) cnt_log, null cnt_sby from v$log group by bytes, thread#
        union all
        select bytes, thread# thr, null cnt_log, count(*) cnt_sby from v$standby_log group by bytes, thread#
        )
        group by bytes, thr
/
  2    3    4    5    6    7
     BYTES        THR       DIFF
---------- ---------- ----------
 104857600          1         -1
 104857600          2          1


db12> begin
for c in (
  select bytes, thr, (max(cnt_log) - nvl(max(cnt_sby), 0)) diff from (
  select bytes, thread# thr, count(*) cnt_log, null cnt_sby from v$log group by bytes, thread#
  union all
  select bytes, thread# thr, null cnt_log, count(*) cnt_sby from v$standby_log group by bytes, thread#
  )
  group by bytes, thr
) loop
  for n in 1..c.diff + 1 loop
    execute immediate 'alter database add standby logfile thread '||c.thr||' size '||c.bytes;
  end loop;
  2    3    4    5    6    7    8    9   10   11   12   13  end loop;
end;
/
 14   15

PL/SQL procedure successfully completed.

Elapsed: 00:00:00.97
db12> db12> select * from  v$log;

    GROUP#    THREAD#  SEQUENCE#      BYTES  BLOCKSIZE    MEMBERS ARC STATUS           FIRST_CHANGE# FIRST_TIM NEXT_CHANGE# NEXT_TIME     CON_ID
---------- ---------- ---------- ---------- ---------- ---------- --- ---------------- ------------- --------- ------------ --------- ----------
         1          1         13  104857600        512          2 YES INACTIVE               1616137 10-JUL-20      1638880 10-JUL-20          0
         2          1         14  104857600        512          2 YES INACTIVE               1638880 10-JUL-20      1659605 10-JUL-20          0
         3          1         15  104857600        512          2 NO  CURRENT                1659605 10-JUL-20   1.8447E+19                    0
         8          2          0  104857600        512          2 YES UNUSED                       0                      0                    0

Elapsed: 00:00:00.01
db12> select * from v$standby_log;

    GROUP# DBID                                        THREAD#  SEQUENCE#      BYTES  BLOCKSIZE       USED ARC STATUS     FIRST_CHANGE# FIRST_TIM NEXT_CHANGE# NEXT_TIME LAST_CHANGE# LAST_TIME     CON_ID
---------- ---------------------------------------- ---------- ---------- ---------- ---------- ---------- --- ---------- ------------- --------- ------------ --------- ------------ --------- ----------
         4 UNASSIGNED                                        1          0  104857600        512          0 YES UNASSIGNED                                                                                0
         5 UNASSIGNED                                        1          0  104857600        512          0 YES UNASSIGNED                                                                                0
         6 UNASSIGNED                                        1          0  104857600        512          0 YES UNASSIGNED                                                                                0
         7 UNASSIGNED                                        1          0  104857600        512          0 YES UNASSIGNED                                                                                0
         9 UNASSIGNED                                        2          0  104857600        512          0 YES UNASSIGNED                                                                                0
        10 UNASSIGNED                                        2          0  104857600        512          0 YES UNASSIGNED                                                                                0

6 rows selected.

Elapsed: 00:00:00.01

Reply:
Got it;  I didn't make the mental leap that a -1 wouldn't show up in the loop.